### ISSUE TITLE
Move privateers to task based

### ISSUE THREAD

**Issue by Dasku** [ScreepsOCS/screeps.behaviour-action-pattern#416]
Privateers should be moved to task based system

Privateers should not travel more than 4 range?

**Comment by Dasku** [comment]
I've been thinking about this a bit, its not a simple spawn a privateer per flag, I guess we need a new macCount calc which takes into account RCL and number of sources, maybe range?, but we don't know the spawn before we try to spawn the creep. And if we did a check for RCL after the creep is made it would still return this as the best room for the flag each loop.

Not sure how we go about this unless we just say 2 privateer per source per room? But at RCL 3, this would slow us down (maybe?), unless we lower the requirements for remote mining to RCL 3 which could solve this issue. That way privateers become only used for special things like preping a room with roads and containers before mining, or stealing from a room other people are mining, or robbing.

**Comment by cyberblast** [comment]
I don't think we should migrate the whole privateer feature to the task system. 
We have that new task based remoteMining system. And in the long run I don't think we need both.

**Comment by Dasku** [comment]
I've now made the robbing task which I think works, just waiting to find a way into someones room (with a full storage) to test it properly before I release it. 

This now free's up the current privateer as mining rooms send workers for construction.

I propose we keep them for now with maybe 1 per flag standard spawn (for red/green flag). They can then be sent to rooms to dismantle walls or or other things in enemy rooms like building roads for haulers (so we can reduce the MOVE parts on the robbers) - but im not sure if this approach is worth it for robbing 200-1000k energy/minerals from a full storage. 

Maybe we just keep the privateer for now and re-develop it into to something completely different, like a scout (1 MOVE and send for vision, could plan mini routes) or some kind of SK room aid creep (to heal miners if injured and SKK during invaders) as has been discussed in other comments.  Or anything really if anyone has any other ideas?

**Comment by cyberblast** [comment]
hmm yes we can keep them for a while. 
But I think for any job, we maybe should develop new specialists.

**Comment by chobobobo** [comment]
Yeah, when I tried to use privateers to dismantle enemy buildings, they did, but then they proceeded to repair all of the surrounding structures!  I don't need my creeps repairing the enemy!  I think it might be time to phase privateers out.

**Comment by karlthepagan** [comment]
I think just another situational flag should prevent repair.

**Comment by chobobobo** [comment]
Now that we have robbing and this is Wont Fix, can it be closed?

**Comment by karlthepagan** [comment]
Unsure.

Privateers are based on a special function in room.js IMO that could be remedied.
